dataverse/webapi/C#-NETx/BulkOperations/Utility.cs

```diff
@@ -272,8 +288,186 @@ public static async Task DeleteExampleTable(Service service, string tableSchemaN
+
+            // TODO: You can use service.SendAsync<RetrieveMultipleResponse>(request) to get 
+            // a typed RetrieveMultipleResponse object returned with a Records property that could
+            //  replace this whole method.
+            var response = await service.SendAsync(request).ConfigureAwait(false);
+
+            JObject jsonResponse = JsonConvert.DeserializeObject<JObject>(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            return jsonResponse;
+        }
+
+        /// <summary>
+		/// Get the response url from Response Message
+		/// </summary>
+		/// <param name="response"></param>
+		/// <returns></returns>
+		private static string GetUrlFromResponse(HttpResponseMessage response)
+        {
+            var responseUrl = response.Headers.GetValues("OData-EntityId").FirstOrDefault().ToString();
+
+            var returnUrl = responseUrl.Substring(responseUrl.IndexOf("EntityDefinitions"));
+
+            return returnUrl;
+        }
+
+        /// <summary>
+		/// Creates alternate key to the entity requested.
+		/// </summary>
+		/// <param name="entitySchemaNane">Entity schema name</param>
+		/// <param name="schemaName">Schema name.</param>
+		/// <param name="displayName">Display name.</param>
+		/// <param name="keyAttributes">List of Key Attributes.
+		/// These attributes should exists in the specified entity.
+		/// These attributes must be of type Decimal or Integer or String.</param>
+		/// <returns>Return True if success</returns>
+		private static async Task<HttpResponseMessage> CreateAlternateKeyToEntityAsync(Service service, 
+            string entitySchemaName, 
+            string schemaName, 
+            string displayName, 
+            List<string> keyAttributes, 
+            bool isSynchronous = false,  //TODO: Remove unused parameters
+            bool isSecondaryKey = false, 
+            HttpStatusCode expectedStatusCode = HttpStatusCode.NoContent)
+        {
+            var entityKeyMetadata = new EntityKeyMetadata
+            {
+                // TODO
+                SchemaName = string.Format("new_{0}", schemaName),
+                LogicalName = string.Format("new_{0}", schemaName),
+                EntityLogicalName = entitySchemaName,
+                KeyAttributes = keyAttributes.ToArray(),
+                DisplayName = new Label(displayName, 1033)
+            };
+
+            // TODO you don't need this method
+            var entityId = await GetEntityMetadataId(service, entitySchemaName).ConfigureAwait(false);
+
+            // TODO: CreateEntityKeyRequest could be refactored to support LogicalName (entitySchemaName.ToLower()) 
+            // instead of using the MetadataId.
+            var createEntityKeyRequest = new CreateEntityKeyRequest(entityKeyMetadata, entityId);
+            // TODO: If you used service.SendAsync<CreateEntityKeyResponse>(createEntityKeyRequest) you should have an 
```

Comment:
If you used service.SendAsync<CreateEntityKeyResponse>(createEntityKeyRequest) you should have an 
CreateEntityKeyResponse.EntityKeyId

Reply:
CreateEntityKey POST API call does not return any content. Return value is not needed here. I have used LogicalName to support retrieval of the EntityKeyMetadata